htdocs/compta/paiement_charge.php

```diff
@@ -223,9 +223,16 @@
 	print '<tr><td class="tdtop">'.$langs->trans("RemainderToPay").'</td><td>'.price($total-$sumpaid,0,$outputlangs,1,-1,-1,$conf->currency).'</td></tr>';*/
 
 	print '<tr><td class="fieldrequired">'.$langs->trans("Date").'</td><td>';
-	$datepaye = dol_mktime(12, 0, 0, $_POST["remonth"], $_POST["reday"], $_POST["reyear"]);
-	$datepayment = empty($conf->global->MAIN_AUTOFILL_DATE) ? (empty($_POST["remonth"]) ?-1 : $datepaye) : 0;
-	print $form->selectDate($datepayment, '', '', '', '', "add_payment", 1, 1);
+	$empty=1;
+	$datepayment='';
+	if (GETPOSTISSET('remonth')) {
+		$datepayment = dol_mktime(12, 0, 0, GETPOST("remonth", 'int'), GETPOST('reday', 'int'), GETPOST('reyear', 'int'));
+		$empty=0;
+	} elseif (!empty($conf->global->MAIN_AUTOFILL_DATE)) {
+		$datepayment = dol_mktime(12, 0, 0, dol_print_date(dol_now(), '%m'), dol_print_date(dol_now(), '%d'), dol_print_date(dol_now(), '%Y'));
+		$empty=0;
+	}
+	print $form->selectDate($datepayment, '', '', '', $empty, "add_payment", 1, 1);
```

Comment:
If a date was posted, the datepayment is retreived with correct date. But param empty is 1 meaning we can clear field to have no data into date instead of 0 (field is required).
What's wrong with old code ?

Reply:
with old code if MAIN_AUTO_FILLDATE = 1 => paiment date is prefilled with 01/01/1970.

Reply:
So why not just replacing
$datepayment = empty($conf->global->MAIN_AUTOFILL_DATE) ? (empty($_POST["remonth"]) ?-1 : $datepaye) : 0;
with
$datepayment = empty($conf->global->MAIN_AUTOFILL_DATE) ? (empty($_POST["remonth"]) ?-1 : $datepaye) : '';
(we use '' to say current date instead of 0 that mean 01/01/1970)